backend/src/services/secret/secret-service.ts

```diff
@@ -776,6 +819,14 @@ export const secretServiceFactory = ({
       subject(ProjectPermissionSub.Secrets, { environment, secretPath: path })
     );
 
+    const isProjectBeingUpgraded = await projectDAL.isProjectBeingUpgraded(projectId);
+
+    if (isProjectBeingUpgraded) {
+      throw new BadRequestError({
+        message: "Project is currently being upgraded, and secrets cannot be written. Please try again"
+      });
+    }
```

Comment:
nit: looks like this check is being repeated a lot, maybe you can put it in a function

Reply:
also hmm i tried to edit secrets while migration was happening and i didn't get any errors